src/Spice86.Core/Emulator/OperatingSystem/DosFileManager.cs

```diff
@@ -673,36 +707,47 @@ public class DosFileManager {
     /// <param name="forCreation">if true will try to find case sensitive match for only the parent of the file</param>
     /// <returns>the file name in the host file system, or null if nothing was found.</returns>
     public string? ToHostCaseSensitiveFileName(string dosFileName, bool forCreation) {
-        string fileName = ToHostFileName(dosFileName);
+        string fileName = ToHostFilePath(dosFileName);
         if (!forCreation) {
-            return ToCaseSensitiveFileName(fileName);
+            string? caseSensitivePath = ToCaseSensitiveFileName(fileName);
+            return caseSensitivePath;
         }
-        string? parent = ToCaseSensitiveFileName(Path.GetDirectoryName(fileName));
-        if (parent == null) {
+        
+        string? parent = ToCaseSensitiveFileName(fileName);
+        if (string.IsNullOrWhiteSpace(parent)) {
             return null;
         }
+        
         // Concat the folder to the requested file name
-        return Path.Combine(parent, dosFileName);
+        return ConvertUtils.ToSlashPath(parent);
     }
 
     /// <summary>
     /// Prefixes the given filename by either the mapped drive folder or the current folder depending on whether there is
     /// a Drive in the filename or not.<br/>
-    /// Does not convert to case sensitive filename.
+    /// Does not convert to case sensitive filename. <br/>
+    /// Does not search for the file or folder on disk.
     /// </summary>
     /// <param name="dosFileName">The file name to convert.</param>
     /// <returns>The converted file name.</returns>
-    private string ToHostFileName(string dosFileName) {
+    private string ToHostFilePath(string dosFileName) {
         string fileName = ConvertUtils.ToSlashPath(dosFileName);
-        if (fileName.Length >= 2 && fileName[1] == ':') {
+        if (IsDosPathRooted(fileName)) {
             fileName = ReplaceDriveWithHostPath(fileName);
-        } else if (string.IsNullOrWhiteSpace(_currentDir) == false) {
-            fileName = Path.Combine(_currentDir, fileName);
+        } else if (string.IsNullOrWhiteSpace(CurrentHostDirectory) == false) {
+            fileName = Path.Combine(CurrentHostDirectory, fileName);
         }
 
         return ConvertUtils.ToSlashPath(fileName);
     }
 
+    /// <summary>
+    /// All the possible DOS drive letters
+    /// </summary>
+    public string[] DriveLetters { get; } = {"A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z" };
+    
+    private bool IsDosPathRooted(string fileName) => fileName.Length >= 2 && DriveLetters.Contains(fileName[0].ToString().ToUpperInvariant()) && fileName[1] == ':';
```

Comment:
Really makes me think all the files and folders conversion / browsing stuff (mapping between host and emulated files) should be in a separate class considering how complex this is becoming

Reply:
This is on-going.

One thing missing is 8.3 file names.

There is another class wanting to get out of the DOS file manager, indeed.

Reply:
DosFIleResolver class created, along with interface.